surfactant/infoextractors/js_file.py

```diff
@@ -4,19 +4,49 @@
 # SPDX-License-Identifier: MIT
 import json
 import re
+from datetime import datetime, timezone
 from typing import Any, Dict, List, Optional
 
 import requests
 from loguru import logger
 
 import surfactant.plugin
 from surfactant.configmanager import ConfigManager
+from surfactant.database_manager.database_utils import (
+    calculate_hash,
+    load_hash_and_timestamp,
+    save_hash_and_timestamp,
+)
 from surfactant.sbomtypes import SBOM, Software
 
 
 class JSDatabaseManager:
```

Comment:
Probably quite a bit here that could be pulled out into a parent class that could also live in the db manager module, that can then be re-used both for the javascript and native library recognition databases.